### PR TITLE
Fix changelog baseline

### PR DESCRIPTION
**Pull request by erwinmombay**
Fixes https://github.com/ampproject/amphtml/issues/20027

using the github API to determine the last branch can sometimes cause bugs like when the last tag is in draft mode or if production patches is the last branch on github

### CODE REVIEW

**Comment by erwinmombay** [comment]
@rsimha @choumx PTAL

File: build-system/tasks/changelog.js [hunk 1]
```diff
@@ -38,8 +38,7 @@ const {GITHUB_ACCESS_TOKEN} = process.env;
 const exec = BBPromise.promisify(childProcess.exec);
 const gitExec = BBPromise.promisify(git.exec);
```
Comment (rsimha) [inline build-system/tasks/changelog.js]: Cleanup task for another PR: If it makes sense, we can consolidate this code with the code in [build-system/git.js](https://github.com/ampproject/amphtml/blob/master/build-system/git.js).

Comment (erwinmombay) [reply build-system/tasks/changelog.js]: sounds good

Comment (erwinmombay) [reply build-system/tasks/changelog.js]: filed here https://github.com/ampproject/amphtml/issues/20138

**Pull request merged**